pkg/server/download.go

```diff
@@ -293,9 +293,11 @@ func (s *server) downloadFileFromPresignedUrl(ctx context.Context, url string, t
 		return 0, err
 	}
 
-	err = os.Rename(tempPath, targetLocation)
+	err = fileSystemSafeRename(tempPath, targetLocation)
+
+	// Catch the error, read in the data from tempPath and write to targetLocation
 	if err != nil {
-		return 0, err
+		log.Fatal("Moving file failed: %v", err)
```

Comment:
might need to wrap in something like `log.Fatal(fmt.Errorf("Moving file failed, %w", err))`